Framework/DataHandling/test/LoadAsciiStlTest.h

```diff
@@ -0,0 +1,86 @@
+#ifndef LOAD_ASCIISTL_TEST_H_
+#define LOAD_ASCIISTL_TEST_H_
+
+#include "MantidAPI/FileFinder.h"
+#include "MantidDataHandling/LoadAsciiStl.h"
+#include "MantidGeometry/Objects/MeshObject.h"
+#include <cxxtest/TestSuite.h>
+
+using namespace Mantid;
+using namespace Mantid::API;
+using namespace Mantid::DataHandling;
+
+using namespace Mantid::Geometry;
+
+class LoadAsciiStlTest : public CxxTest::TestSuite {
+public:
+  static LoadAsciiStlTest *createSuite() { return new LoadAsciiStlTest(); }
+  static void destroySuite(LoadAsciiStlTest *suite) { delete suite; }
+
+  void assert_shape_matches(std::unique_ptr<Geometry::MeshObject> &shape,
+                            int vertices, int triangles, int volume,
+                            double delta) {
+    TS_ASSERT(shape->hasValidShape());
+    TS_ASSERT_EQUALS(shape->numberOfVertices(), vertices);
+    TS_ASSERT_EQUALS(shape->numberOfTriangles(), triangles);
+    TS_ASSERT_DELTA(shape->volume(), volume, delta);
+  }
+
+  void test_cube() {
+    std::string path = FileFinder::Instance().getFullPath("cube.stl");
+    std::unique_ptr<LoadAsciiStl> Loader = std::make_unique<LoadAsciiStl>(path);
+    auto cube = Loader->readStl();
+
+    assert_shape_matches(cube, 8, 12, 3000, 0.001);
+  }
+
+  void test_cylinder() {
+    std::string path = FileFinder::Instance().getFullPath("cylinder.stl");
+    std::unique_ptr<LoadAsciiStl> Loader = std::make_unique<LoadAsciiStl>(path);
+    auto cylinder = Loader->readStl();
+
+    assert_shape_matches(cylinder, 722, 1440, 589, 1);
+  }
+
+  void test_tube() {
+    std::string path = FileFinder::Instance().getFullPath("tube.stl");
+    std::unique_ptr<LoadAsciiStl> Loader = std::make_unique<LoadAsciiStl>(path);
+    auto tube = Loader->readStl();
+
+    TS_ASSERT(tube->hasValidShape());
+    assert_shape_matches(tube, 1080, 2160, 7068, 1);
+  }
+
+  void test_fail_invalid_stl_keyword() {
+    loadFailureTest("invalid_keyword.stl");
+  }
+
+  void test_fail_invalid_stl_vertex() { loadFailureTest("invalid_vertex.stl"); }
+
+  void test_fail_invalid_stl_triangle() {
+    loadFailureTest("invalid_triangle.stl");
+  }
+
+  void loadFailureTest(const std::string filename) {
+    std::string path = FileFinder::Instance().getFullPath(filename);
+    std::unique_ptr<LoadAsciiStl> Loader = std::make_unique<LoadAsciiStl>(path);
+
+    TS_ASSERT_THROWS_ANYTHING(Loader->readStl());
+  }
+
+  // check that isAsciiSTL returns false when loading a binary
```

Comment:
The test method name should be descriptive enough without additional comments, or that's what our guidelines say.